add OSS CTA for auth connectors

michellescripts opened the pull request:
This PR adds an OSS CTA for Auth Connectors.


<img width="1693" alt="Screenshot 2023-08-24 at 11 59 32 AM" src="https://github.com/gravitational/teleport/assets/11967646/26428c52-1594-49b2-896f-5a05a1ff521e">

<img width="1240" alt="Screenshot 2023-08-24 at 11 59 38 AM" src="https://github.com/gravitational/teleport/assets/11967646/7c7db0ba-b006-4153-b3b1-99575c7d88a0">

<img width="651" alt="Screenshot 2023-08-24 at 11 59 48 AM" src="https://github.com/gravitational/teleport/assets/11967646/a3666872-1105-43cd-8a55-c1f62f3cf788">



Moved code from E into OSS to alleviate duplication; removed/editied in: https://github.com/gravitational/teleport.e/pull/2026

Supports https://github.com/gravitational/cloud/issues/5598

michellescripts committed:
add cta for auth conn

avatus commented on web/packages/teleport/src/AuthConnectors/AuthConnectors.tsx:
```diff
@@ -71,48 +78,50 @@ export function AuthConnectors(props: State) {
+              onDelete={resources.remove}
+            />
+            <Box
+              ml="4"
```
```suggestion
              ml={4}
```

avatus commented on web/packages/teleport/src/AuthConnectors/EmptyList/EmptyList.tsx:
```diff
@@ -16,47 +16,102 @@ limitations under the License.
+        flexWrap="wrap"
+        style={{ position: 'relative' }}
+        justifyContent="center"
+        mt="4"
```
```suggestion
        mt={4}
```

avatus commented on web/packages/teleport/src/AuthConnectors/EmptyList/EmptyList.tsx:
```diff
@@ -16,47 +16,102 @@ limitations under the License.
-          Open Source Teleport supports only GitHub connectors. Please{' '}
+    <Card
+      color="text.main"
+      p="5"
```
```suggestion
      p={5}
```

avatus reviewed (commented):
I'd like to keep the margin/padding values as number instead of a mix of number/string. (happy for pushback if we are moving towards string tho?)

From the screenshot, the button looks to be in a bit of a weird place. Is there a chance we could center it vertically (and horizontal?) between the two faded options?

ryanclark commented on web/packages/teleport/src/AuthConnectors/AuthConnectors.tsx:
```diff
@@ -71,48 +78,50 @@ export function AuthConnectors(props: State) {
+              style={{ flexShrink: 0 }}
+              css={`
+                @media screen and (max-width: 1000px) {
+                  display: none;
```
Instead of hiding this when the screen is too small, can it wrap to underneath the `ConnectorList`?

ryanclark commented on web/packages/teleport/src/AuthConnectors/AuthConnectors.tsx:
```diff
@@ -52,13 +52,20 @@ export function AuthConnectors(props: State) {
     <FeatureBox>
-      <FeatureHeader>
+      <FeatureHeader
+        css={`
+          @media screen and (max-width: 800px) {
+            flex-direction: column;
+            height: auto;
+            gap: 10px;
+            margin: 0 0 10px 0;
+            padding-bottom: 10px;
+            justify-content: center;
+          }
+        `}
```
Would 

```
const ResponsiveFeatureHeader = styled(FeatureHeader)`
  @media screen and (max-width: 800px) {
    flex-direction: column;
    height: auto;
    gap: 10px;
    margin: 0 0 10px 0;
    padding-bottom: 10px;
    justify-content: center;
  }
`;
```

be better here? I don't have a strong preference, I just try to avoid having `css` blocks in the layout

ryanclark commented on web/packages/teleport/src/AuthConnectors/styles/ConnectorBox.styles.ts:
```diff
@@ -0,0 +1,49 @@
+  transition: all 0.3s;
+  border-radius: 4px;
+  min-height: 190px;
+  border: 2px solid ;
```
Is this missing a color?

ryanclark commented on web/packages/teleport/src/AuthConnectors/styles/ConnectorBox.styles.ts:
```diff
@@ -0,0 +1,49 @@
+    box-shadow: ${p => p.theme.boxShadow[3]};
+    cursor: pointer;
+  }
+  &:disabled {
+    cursor: not-allowed;
+  color: inherit;
+  font-family: inherit;
+  outline: none;
+  position: relative;
+  text-align: center;
+  text-decoration: none;
+  &:disabled {
+    opacity: .24;
+    box-shadow: none;
+  }
```
There's two `&:disabled` here and it doesn't look like the first `&:disabled`'s brace is closed

michellescripts committed:
responsive & feedback

michellescripts commented:
@avatus 

> I'd like to keep the margin/padding values as number instead of a mix of number/string. (happy for pushback if we are moving towards string tho?)

Nope, these were just copied/moved from E so I didn't really look at them, updated.

>From the screenshot, the button looks to be in a bit of a weird place. Is there a chance we could center it vertically (and horizontal?) between the two faded options?

Fixed

@ryanclark 

> Instead of hiding this when the screen is too small, can it wrap to underneath the ConnectorList?

As per design, I put it at the top
